tests/integration/security/util/headers/builder.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package headers
+
+import (
+	"net/http"
+
+	"istio.io/istio/pkg/test/echo/common"
+	epb "istio.io/istio/pkg/test/echo/proto"
+)
+
+// Builder for HTTP headers.
+type Builder struct {
+	headers http.Header
+}
+
+// New Builder for HTTP headers.
+func New() *Builder {
+	return &Builder{
+		headers: make(http.Header),
+	}
+}
+
+// With sets the given header value.
+func (b *Builder) With(key, value string) *Builder {
+	b.headers.Set(key, value)
+	return b
+}
+
+// WithAuthz sets the Authorization header with the given token.
+func (b *Builder) WithAuthz(token string) *Builder {
+	if token != "" {
+		return b.With(Authorization, "Bearer "+token)
+	}
+	return b
+}
+
+// WithHost sets the Host header to the given value.
+func (b *Builder) WithHost(host string) *Builder {
+	return b.With(Host, host)
+}
+
+// WithXForwardedFor sets the origin IP of the request.
+func (b *Builder) WithXForwardedFor(ip string) *Builder {
+	return b.With(XForwardedFor, ip)
+}
+
+func (b *Builder) Build() http.Header {
```

Comment:
can you make `type Builder http.Header` instead of a struct, then avoid the need for Build?

Reply:
By doing that, we lose the ability to call the http.Header methods, like Set/Get.

Reply:
bummer

Reply:
How about this:

```go
// Builder for HTTP headers.
type Builder http.Header

// With sets the given header value.
func (b Builder) With(key, value string) Builder {
	http.Header(b).Set(key, value)
	return b
}
func TestT(t *testing.T) {
	t.Log(Builder{}.With("foo", "bar"))
}
```

Reply:
Oh then its not a http.Header though so you still cannot pass it to things that want http.Header. lame